src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -216,7 +216,7 @@ internal DeltaSet<TStructuralType> CopyChangedValues(IODataAPIHandler apiHandler
                         {
                             operation = DataModificationOperationKind.Update;
 
-                            ODataAPIResponseStatus linkResponseStatus = handler.TryAddRelatedObject(original, out errorMessage);
+                            ODataAPIResponseStatus linkResponseStatus = apiHandlerOfT.TryAddRelatedObject(original, out errorMessage);
```

Comment:
Line 110 we assign `apiHandlerOfT` to handler. Why are the different at this point?

Reply:
I explained that in the PR description.

In line 149, we assign the `handler` to `odataPathApiHandler`. This will cause issues in line 219

Reply:
Ah I see.

Reply:
I think we should do a check..if `odataApiHandler` is null then we should use the `handler`, otherwise then we use `apiHandlerOfT` but I guess even doing it this way is fine.